### PR TITLE
Add `force_system_arguments` options instead of relying on RAILS_ENV

### PR DESCRIPTION
**Pull request by manuelpuyol**
Adds a configuration that allows users to choose if not using System Arguments should raise an error or not. It will allow users to select which environments will raise an error.

### CODE REVIEW

File: app/lib/primer/classify.rb [hunk 1]
```diff
@@ -236,6 +236,10 @@ def extract_value(memo, key, val, breakpoint)
           memo[:classes] << "#{key.to_s.dasherize}#{breakpoint}-#{val.to_s.dasherize}"
         end
       end
+
+      def force_system_arguments?
+        Rails.application.config.primer_view_components.force_system_arguments
```
Comment (joelhawksley) [inline app/lib/primer/classify.rb:241]: Can we explicitly set this value to false by default? We wouldn't want this to accidentally be turned on in production.

File: lib/primer/view_components/engine.rb [hunk 2]
```diff
@@ -13,7 +13,9 @@ class Engine < ::Rails::Engine
       ]
 
       config.primer_view_components = ActiveSupport::OrderedOptions.new
+
       config.primer_view_components.force_functional_colors = true
+      config.primer_view_components.force_system_arguments = true
```
Comment (BlakeWilliams) [inline lib/primer/view_components/engine.rb:18]: Would it make sense to keep the default value based on the environment?

Maybe `config.primer_view_components.force_system_arguments = !Rails.env.production?`

**Review by BlakeWilliams — APPROVED** [review]
LGTM!

File: lib/primer/view_components/engine.rb (hunk 2 above)
Comment (manuelpuyol) [reply lib/primer/view_components/engine.rb]: I'll leave it false by default and let users decide

**Pull request merged**